### DISCUSSION TITLE
Issue with running select query in spatial extension

### DISCUSSION THREAD

**Discussion by przygocka** [duckdb/duckdb-spatial#167] [Q&A] [answered]
Hello,
I am playing around with duckdb_spatial extension and I encountered a problem with running following query: Select * from zones WHERE ST_Area(geom) > 1000000;

I loaded the example data taxi_zones.shx and I got a following error: runtime error: member access within misaligned address 0x7f849824e03c for type 'spatial::core::Vertex', which requires 8 byte alignment
0x7f849824e03c: note: pointer points here
  00 00 00 00 80 ff b7 7c  6a 88 2f 41 00 00 00 10  f0 02 05 41 80 ff 07 49  bf 89 2f 41 00 04 00 91
              ^ 
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /Users/duckdb_spatial/spatial/src/spatial/core/geometry/vertex_vector.cpp:70:12 in 

I tried installing and loading spatial extension, running the same query on the same dataset but in the original duckdb code (https://github.com/duckdb/duckdb) and the query worked. 

Does anybody know why is this happening, am I doing smth wrong in duckdb_spatial extension?

**Comment by Maxxen** [comment, marked as answer]
Hi! When building spatial in debug mode, UBSAN - a tool to detect potential errors is enabled. I'm aware of this issue, In practice it's not a problem but it hasn't been fixed yet since I'm reworking this part of the code anyway. However, this is a good reminder to finish it sooner rather than later..

**Comment by Maxxen** [comment]
Give me a couple of hours and Ill squash this
